internal/cli/root/builder.go

```diff
@@ -247,6 +207,57 @@ Use the --help flag with any command for more info on that command.`,
 		federatedauthentication.Builder(),
 	)
 
+	// create copy of the subcommands slice before adding the commands from the plugins
+	// That allows us to differentiate between normal subcommands and plugin subcommands
+	// and we can use it to set all normal subcommands to RequireAuth
+	// and all plugin subcommands to OptionalAuth in shouldCheckCredentials
+	subCommandsWithoutPlugins := make([]*cobra.Command, len(rootCmd.Commands()))
```

Comment:
Are these changes needed only for identifying commands from plugins vs commands from cli and use this info in `shouldCheckCredentials`?
If yes, wdyt about using annotations? e.g. when registering the plugin commands you can add cmd.Annotations["sourceType"] = "plugin" and cmd.Annotations["sourceName"] = "k8s".
Then in shouldCheckCredentials you can check cmd.Annotations["sourceType"]

Reply:
done